### PR TITLE
Homepage revamp

### PR DESCRIPTION
**Pull request by itexpert120**

### CODE REVIEW

File: src/home/components/Connect.jsx [hunk 1]
```diff
@@ -0,0 +1,232 @@
+State.init({
```
Comment (elliotBraem) [inline src/home/components/Connect.jsx:1]: You can use useState now

File: src/home/components/Banner.jsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
+const imageLink =
+  "https://s3-alpha-sig.figma.com/img/b1f8/9258/16e820ecd12b0c541cdd7177d0b53a24?Expires=1698624000&Signature=owRGoRIMd35cUHfLyMvuYirlRCAdUoTn801J3ozXJRgFp~Pfy9yLCklGK~wigQK32HTJ6YET6QbAjeP20YEO5NE5ymAL9V3LBJGkVA-X-AcnZATk3QAEzM5S5cmmlFlzf~r4DLiKs5vFKPYYVUptoYlSFqmBIGGDNaBfdl5WvTLQqCXUyw62oHlsEi8RX3mO99L-UdttIpZHeJvJBBCj21HUzM461rULCxwkXBp6mvqmBq6kF0lhXDH4PcwVkiuzqRDSu0myL2hcaUY5QpFtm8tOuh0z-FMNtdImxqVtbH3CbD9J7xsOeFcpkAvQZFDRgtAAlnto0-lfASsWaioXSQ__&Key-Pair-Id=APKAQ4GOSFWCVNEHN3O4";
```
Comment (elliotBraem) [inline src/home/components/Banner.jsx:2]: Replace with this:
https://ipfs.near.social/ipfs/bafybeiap2mzwsly4apaldxguiunx4rjwqyadksj5yxuzwrww3kue3ao5qe

File: src/home/components/Connect.jsx [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+  color: #f4f4f4;
+  text-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
+  font-size: 20px;
+  font-style: normal;
+  font-weight: 400;
+  line-height: 120%; /* 28.8px */
+`;
+
+const imageSource =
+  "https://s3-alpha-sig.figma.com/img/3b54/e4fa/1508a7cf222faf5f331f489d224e4fd6?Expires=1698624000&Signature=gIhYLppLobWi6QMrBzxH-SQFd5JGKJrah~PHlxUYgOOoYPxOTTQ9qOEB8xqHX8pGuaJqTre7Fn~FdI~jWK0WdPrvev8OJDIKWF1zSLSLYG20ztuMR8K8hu4~N1k-Fomyp34H6GalsbWM6p~J5QS0G5L-Wkc2am24-Ys8c4BwqH-CdPAvehTNxgwgGq77aLkf1HqfRaUW5uLcLBhzOavk03lXSxSTmUo5KZX5SOxYTpktmLozASkkFR4-SAVQkxEwlN8s3JG8eX7dsUohsfiE-jYFjuKxfwU~f0OfyvNCvdyvEg335qIMRvD5NCrmQTnm0ZJRxOX~imYi6GV74U8p8w__&Key-Pair-Id=APKAQ4GOSFWCVNEHN3O4";
```
Comment (elliotBraem) [inline src/home/components/Connect.jsx:24]: https://ipfs.near.social/ipfs/bafkreidwtqu7wrppkvwb2criwgk6u7bfx7ojyaempio7pnkhvspxxeujke

Btw, I'm using this:
https://near.social/cuhoangng.near/widget/ImageUploader

File: src/home/components/Connect.jsx [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+    </div>
+    <div style={{ padding: 48, paddingTop: 0 }}>
+      <CTA href="#">Explore all communities →</CTA>
+    </div>
+  </>
+);
+
+return (
+  <Widget
+    src="devhub.testnet/widget/home.components.Section.HomeSection"
```
Comment (elliotBraem) [inline src/home/components/Connect.jsx:225]: Replace with "${REPL_DEVHUB}/widget/home.components.Section.HomeSection"

Use this pattern for all these so it injects properly according to the network

File: src/home/components/Contribute.jsx [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+        </div>
+      ))}
+    </Container>
+    <CTA href="#">Learn more →</CTA>
+  </div>
+);
+
+return (
+  <Widget
+    src="devhub.testnet/widget/home.components.Section.HomeSection"
```
Comment (elliotBraem) [inline src/home/components/Contribute.jsx:97]: same

File: src/home/components/Hero.jsx [hunk 6]
```diff
@@ -0,0 +1,79 @@
+const imageLink =
+  "https://s3-alpha-sig.figma.com/img/b1f8/9258/16e820ecd12b0c541cdd7177d0b53a24?Expires=1698624000&Signature=owRGoRIMd35cUHfLyMvuYirlRCAdUoTn801J3ozXJRgFp~Pfy9yLCklGK~wigQK32HTJ6YET6QbAjeP20YEO5NE5ymAL9V3LBJGkVA-X-AcnZATk3QAEzM5S5cmmlFlzf~r4DLiKs5vFKPYYVUptoYlSFqmBIGGDNaBfdl5WvTLQqCXUyw62oHlsEi8RX3mO99L-UdttIpZHeJvJBBCj21HUzM461rULCxwkXBp6mvqmBq6kF0lhXDH4PcwVkiuzqRDSu0myL2hcaUY5QpFtm8tOuh0z-FMNtdImxqVtbH3CbD9J7xsOeFcpkAvQZFDRgtAAlnto0-lfASsWaioXSQ__&Key-Pair-Id=APKAQ4GOSFWCVNEHN3O4";
+
```
Comment (elliotBraem) [inline src/home/components/Hero.jsx:3]: upload and replace

**Review by elliotBraem — COMMENTED** [review]
You a real pro @itexpert120

File: src/home/components/Connect.jsx (hunk 1 above)
Comment (itexpert120) [reply src/home/components/Connect.jsx:1]: Cool, do we have docs?

File: src/home/components/Banner.jsx (hunk 2 above)
Comment (itexpert120) [reply src/home/components/Banner.jsx:2]: I will

File: src/home/components/Connect.jsx (hunk 3 above)
Comment (itexpert120) [reply src/home/components/Connect.jsx:24]: Thanks for telling me

**Comment by itexpert120** [comment]
resolved all issues @elliotBraem 
Also created a root app layout

**Comment by elliotBraem** [comment]
@itexpert120, could you please rebase these changes against [develop](https://github.com/near/neardevhub-widgets/tree/develop) branch?

This is the current structure in develop branch:

```
├── app.jsx
├── core
│   ├── adapter
│   └── lib
├── devhub
|   ├── components
│   │   ├── atom
│   │   ├── molecule
│   │   ├── organism
│   │   └── templates
│   ├── entity
│   └── page
```

To integrate, this is what I would recommend:

1. Move the home.jsx under devhub/page
2. In app.jsx, replace the route with this home Widget, make sure to passProps:
```javascript
  switch (routes[0]) {
    // ?page=home
    case "home": {
      return <p>Homepage</p>;
    }
```

3. In app.jsx, define any global styles or fonts in the Theme styled component. This will ensure all children have it.
4. Modify the devhub/components/templates/AppLayout to match your layout. You can replace the current code.
5. Components are currently following Atomic design, which has grown me, but I'm also open to whatever makes the most sense. Following atomic design, as I understand it:
* banner, connect, contribute, etc. would go in devhub/components/island since they are completely independent elements. Home-section would be here too?
 * Dropdown-menu looks very reusable (nice) and can be ever expanding; I think this is a devhub/components/organism? Same with the navbar


Also, @itexpert120  I think I fixed the hacks you needed to do the yarn dev script, lmk

@jaswinder6991

**Pull request closed without merging**